picocom.c

```diff
@@ -1540,12 +1540,18 @@ loop(void)
         }
 
         if ( FD_ISSET(tty_fd, &wrset) ) {
+            int have_esc = 0;
+            int sz;
 
             /* write to port */
-            int sz;
             sz = (tty_q.len < tty_write_sz) ? tty_q.len : tty_write_sz;
             do {
-                if (tty_sleep.tv_sec | tty_sleep.tv_nsec) {
+
+                /* Escape sequences shall be delivered directly */
+                if (*tty_q.buff == '\x1b')
+                        have_esc = 1;
+
+                if (!have_esc && (tty_sleep.tv_sec | tty_sleep.tv_nsec)) {
```

Comment:
I think the comment added to the code is good and clear, so I'd suggest to remove the have_esc variable in this case, and just use 'buff != 0x1b' to be more concise.. And squash the commit into the previous one.